wp_api_integration_tests/tests/test_login_remote.rs

```diff
@@ -0,0 +1,443 @@
+use serial_test::parallel;
+use std::{path::Path, sync::Arc};
+use wp_api::{
+    InvalidSslErrorReason, RequestExecutionError, RequestExecutionErrorReason,
+    login::{
+        login_client::WpLoginClient,
+        url_discovery::{
+            ApplicationPasswordsNotSupportedReason, AutoDiscoveryAttemptFailure,
+            FetchAndParseApiRootFailure, FindApiRootFailure,
+        },
+    },
+    middleware::{
+        ApiDiscoveryAuthenticationMiddleware, RetryAfterMiddleware, WpApiMiddleware,
+        WpApiMiddlewarePipeline,
+    },
+    request::RequestExecutor,
+    reqwest_request_executor::ReqwestRequestExecutor,
+};
+use wp_api_integration_tests::mock::{MockExecutor, response_helpers};
+
+#[tokio::test]
+#[parallel]
+async fn login_spec_1_valid_site_works_correctly() {
+    // Spec Example 1
+    assert_eq!(
+        login_url("https://vanilla.wpmt.co").await,
+        "https://vanilla.wpmt.co/wp-admin/authorize-application.php"
+    );
+}
+
+#[tokio::test]
+#[parallel]
+async fn login_spec_2_local_development_environment() {
+    // Spec Example 2
+    let executor = MockExecutor::with_execute_fn(|request| match request.url().0.as_str() {
+        "http://localhost/" | "https://localhost/" => {
+            Ok(response_helpers::with_api_root("http://localhost/wp-json"))
+        }
+        "http://localhost/wp-json" | "https://localhost/wp-json" => {
+            Ok(response_helpers::json_response_from_path(Path::new(
+                "../native/swift/Tests/wordpress-api/Resources/Responses/localhost-json-root.json",
```

Comment:
Kotlin, Swift & Rust tests are using this path. We should move it to a common place, but I wanted to separate that into its own PR.